### PR TITLE
feat(template): add new jinja filters

### PR DESCRIPTION
**Pull request by sileht**
* markdownify
* get_section("### Commit message")

Fixes MRGFY-635

### CODE REVIEW

File: docs/source/configuration.rst [hunk 1]
```diff
@@ -318,12 +318,37 @@ will render to:
...
 .. note::
 
    You need to replace the ``-`` character by ``_`` from the :ref:`pull request
    attribute <attributes>` names when using templates. The ``-`` is not a valid
    character for variable names in Jinja2 template.
 
+.. note::
+
+   By default, the html comment is stripped from ``body``, to get the
+   full body, you should use ``body-raw``.
```
Comment (jd) [inline docs/source/configuration.rst:350]:
```suggestion
   By default, the HTML comments are stripped from ``body``. To get the
   full body, you can use the ``body_raw`` attribute.
```

File: mergify_engine/context.py [hunk 2]
```diff
@@ -1601,7 +1603,13 @@ class PullRequest(BasePullRequest):
     }
 
     async def __getattr__(self, name: str) -> ContextAttributeType:
-        return await self.context._get_consolidated_data(name.replace("_", "-"))
+        data_name = name
+        if name == "body-raw":
+            data_name = "body"
+        data = await self.context._get_consolidated_data(data_name.replace("_", "-"))
+        if name == "body":
+            return re.sub("(<!--.*?-->)", "", typing.cast(str, data))
```
Comment (jd) [inline mergify_engine/context.py:1611]: I'd do it from `_get_consolidate_data`, no? sounds cleaner

File: mergify_engine/context.py [hunk 3]
```diff
@@ -1633,6 +1642,33 @@ async def render_template(
...
+        env.filters["get_section"] = self._filter_get_section
+
+    @staticmethod
+    def _filter_get_section(v: str, section: str) -> str:
+        if not isinstance(section, str):
+            raise TypeError("level must be a string")
+
+        section_escaped = re.escape(section)
+        level = MARKDOWN_TITLE_RE.match(section)
+        print(section_escaped)
```
Comment (jd) [inline mergify_engine/context.py:1656]:
```suggestion
```

File: mergify_engine/context.py [hunk 4]
```diff
@@ -1633,6 +1642,33 @@ async def render_template(
...
+
+        level_str = level[0].strip()
+
+        level_re = re.compile(fr"^{level_str} +", re.I)
+        section_re = re.compile(fr"^{section_escaped}\s*$", re.I)
+
+        lines = v.split("\n")
+        lines = list(itertools.dropwhile(lambda l: not section_re.match(l), lines))
+        if not lines:
+            raise TypeError("section not found")
```
Comment (jd) [inline mergify_engine/context.py:1668]:
```suggestion
        lines = itertools.dropwhile(lambda l: not section_re.match(l), lines)
        if not any(lines):
            raise TypeError("section not found")
```

**Review by jd — CHANGES_REQUESTED** [review]

File: mergify_engine/context.py (hunk 2 above)
Comment (sileht) [reply mergify_engine/context.py:1611]: `_get_consolidate_data()` is also used with the `rules.filter` code. If we change the body there, rules trying to match something in a comment will fail. But maybe we don't care.

Comment (jd) [reply mergify_engine/context.py:1611]: I don't think we care indeed.
We might want to update the doc to include both `body` and `body-raw` though.

**Review by jd — CHANGES_REQUESTED** [review]
release note?

**Review by jd — APPROVED** [review]

**Pull request merged**